src/hip_module.cpp

```diff
@@ -1144,3 +1144,32 @@ hipError_t hipOccupancyMaxActiveBlocksPerMultiprocessorWithFlags(
     return ihipLogStatus(ihipOccupancyMaxActiveBlocksPerMultiprocessor(
         tls, numBlocks, f, blockSize, dynSharedMemPerBlk));
 }
+
+hipError_t hipLaunchKernel(
+    const void* func_addr, dim3 numBlocks, dim3 dimBlocks, void** args,
+    size_t sharedMemBytes, hipStream_t stream)
+{
+   HIP_INIT_API(hipLaunchKernel,func_addr,numBlocks,dimBlocks,args,sharedMemBytes,stream);
+
+   hipFunction_t kd = hip_impl::get_program_state().kernel_descriptor((std::uintptr_t)func_addr,
+                                                           hip_impl::target_agent(stream));
+   size_t szKernArg = 0;
+
+   if(kd != nullptr && kd->_header != nullptr)
+       szKernArg = kd->_header->kernarg_segment_byte_size;
+
+   void* config[]{
+        HIP_LAUNCH_PARAM_BUFFER_POINTER,
+        args,
+        HIP_LAUNCH_PARAM_BUFFER_SIZE,
+	    &szKernArg,
+        HIP_LAUNCH_PARAM_END};
+
+        if(args == NULL)
+        {
+           config[1] = NULL;
```

Comment:
Is this required? If args is NULL, then config[1] should already be set to NULL as per line 1163. Also it might be useful to check if args is NULL but szKernArg is not 0 and return an error appriopriately.